mirny.py

```diff
@@ -310,23 +345,44 @@ def __init__(self, platform):
         clk = platform.request("clk")
         clk_div = TSTriple()
         self.specials += clk_div.get_tristate(clk.div)
-        # in_sel: 00: XO, 01: n/a (SMA+XO), 10: MMXC, 11: SMA
+        # in_sel: 00: XO, 01: n/a (SMA+XO), 10: MMCX, 11: SMA
         # dividers: 00(z): 1, 01(z): 1, 10(low): 2, 11(high) 4
         self.comb += [
             Cat(clk.in_sel, clk_div.o, clk_div.oe).eq(regs[1].write[4:8]),
             platform.request("fsen").eq(~regs[1].write[9]),
         ]
 
-        for i, m in enumerate(platform.request("mezz_io")):
-            tsi = TSTriple()
-            self.specials += tsi.get_tristate(m)
-            self.comb += [
-                tsi.o.eq(regs[3].write[i] | (0 if i >= 4 else
-                    (regs[1].write[11] & eem[i + 4].i))),
-                regs[3].read[i].eq(tsi.i),
-                tsi.oe.eq(regs[3].write[i + 8]),
-                regs[3].read[i + 8].eq(tsi.oe),
-            ]
+        if legacy_almazny:
+            # 6 signals
+            almazny_io = platform.request("legacy_almazny_io")
+            almazny_adr = 0b1100  # 1100 - and then 1101, 1110, 1111 for sr 1-4
+            almazny_mask = 0b0011
+            self.sr.connect_almazny_passthrough(almazny_io, almazny_adr, almazny_mask)
+
+            # 7th, connecting all NOEs
+            for i in range(4):
+                pin = platform.request("legacy_almazny_noe", i)
+                tsi = TSTriple()
+                self.specials += tsi.get_tristate(pin)
```

Comment:
Why can't REG_NOE be permanently asserted?
In any case you don't need a tristate here.

Reply:
Figured it would be better to give some control over output enable, even if it's for all four outputs enable, than to have it enabled at all times.

Sorry if it seems like an amateurish question then, but I did follow the previous code - why was tristate necessary there and not here?

Reply:
The previous code was generic GPIO, this code is specific to Almazny.

Reply:
I also can't see what additional control it gives you. What's the scenario here?